Updated project structure

nilols opened the pull request:
No code changes only the project structure has been transformed into a multi-module maven project.

graphwalker-parent - handles dependencies and plugins versioning, also gives all inheriting modules basic functionality like different kind of report generation. 

graphwalker-core - contains the GraphWalker code, this module can later on be split up into several modules eg. gui and webservice modules.

graphwalker-standalone - builds the executable standalone jar file, the reason for this being done in a separate module is due to the fact that each maven project only should have one build artifact. 

Run "mvn clean install", in each module to compile and install the artifact generated by the module in the local repository, add -Dmaven.test.skip to omit the tests. This is true for all modules except graphwalker-parent, who will instead build the entire project.

KristianKarl-zz commented:
Jag har skapat en development
branchhttps://github.com/KristianKarl/GraphWalker/tree/development
och
lagt det nya dr.
/K

nilols commented:
Tja

Jag tog ut en ny fork I helgen, vet inte om du såg det, eftersom det blev lite för många ändringar på en gång på den förra så det var inte bakåtkompatibelt...vilket känns lite tråkigt.

Sent from my HTC

----- Reply message -----
From: "Kristian Karl" reply@reply.github.com
Date: Mon, Oct 17, 2011 13:11
Subject: [GraphWalker] Updated project structure (#2)
To: "Nils Olsson" Nils.Olsson@sylog.se

Jag har skapat en development
branchhttps://github.com/KristianKarl/GraphWalker/tree/development
och
lagt det nya dr.
/K

## 

Reply to this email directly or view it on GitHub:
https://github.com/KristianKarl/GraphWalker/pull/2#issuecomment-2427500

KristianKarl-zz commented:
De menar att du gjort mer ndringar n de som kom in nu? Kan vi inte merga
in dom nd? Jag hjlper grna till!

2011/10/17 Nils Olsson <
reply@reply.github.com>

> Tja
> 
> Jag tog ut en ny fork I helgen, vet inte om du sg det, eftersom det blev
> lite fr mnga ndringar p en gng p den frra s det var inte
> baktkompatibelt...vilket knns lite trkigt.
> 
> Sent from my HTC
> 
> ----- Reply message -----
> From: "Kristian Karl" reply@reply.github.com
> Date: Mon, Oct 17, 2011 13:11
> Subject: [GraphWalker] Updated project structure (#2)
> To: "Nils Olsson" Nils.Olsson@sylog.se
> 
> Jag har skapat en development
> branchhttps://github.com/KristianKarl/GraphWalker/tree/development
> och
> lagt det nya dr.
> /K
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/KristianKarl/GraphWalker/pull/2#issuecomment-2427500
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/KristianKarl/GraphWalker/pull/2#issuecomment-2428337

Commit:
Added a new ViewAction that will be used to display the result and execution time for all jobs in a view

Commit:
Updated the refresh cycle and graphics

Commit:
Updated graphics

Commit:
Updated graphics

Commit:
Added spinner (spin.js)

Commit:
We need to check if any errors occurred before updating the model status, otherwise the model can get the wrong status (e.g. when using a exception strategy that never fail the execution of a model)

Commit:
Added a new exception strategy

Commit:
Fixed a problem regarding reporting of test results

Commit:
Updated plugin classes

Commit:
Updated jenkins plugin

Commit:
When using the -Dtest=MyImplementation parameter and if MyImplementation is not found, the graphwalker plugin will fail due to a IllegalArgumentException. The exception is thrown when the plugin try to execute "nothing", a check is added to make sure we have something to execute before we do it.

Commit:
Renamed modules to conform to maven standard

Commit:
Renamed modules to conform to maven standard

Commit:
Updated readme

Commit:
Added the intellij plugin as a module

Commit:
Updated the jenkins plugin.

Commit:
Updated the intellij plugin

Commit:
Added a new jira module

Commit:
New StopCondition that always call a user defined method

Commit:
update

Commit:
Removed unused modules and changed the package of the status classes

Commit:
Moved implementations into impl subpackages

Commit:
refactor

Commit:
Removed intellij plugin naming convention of variables

Commit:
Refactor

Commit:
Added embryo for "Eating your own dog food" tests

Commit:
Changed the internal model to a immutable version, and also created a ModelContext that will be used when executing a model.

Commit:
Changed the internal model to a immutable version, and also created a ModelContext that will be used when executing a model.

Commit:
Changed the internal model to a immutable version, and also created a ModelContext that will be used when executing a model.

Commit:
New modules

Commit:
Added thrift definition file

Commit:
Added more functions to the graphwalker thrift service and improved the server implementation

Commit:
Improved the api

Commit:
Improved the api

Commit:
Added classes for the shopping cart example

Commit:
Added pom file for the graphwalker-example

Commit:
ExecutionContext defines all models known to the context, also added fulfillment variable.

Commit:
Removed the Algorithm interface, because its not used

Commit:
Removed the Report generation and Callback condition from core module

Commit:
Removed BacktrackStrategy and moved the GraphWalker java object to the java module

Commit:
Wraps the GraphWalkerService.Client object into a GraphWalker object, all calls are delegated to the client

Commit:
Removed the Configuration class, ModelContext is used instead

Commit:
Refactor

Commit:
Refactor

Commit:
Added implementation for stop conditions and path generators

Commit:
Executes actions on edges and added annotations

Commit:
Added ExceptionHandler annotation and updated the Amazon example to use annotations for handling the browser object.

Commit:
Rename method

Commit:
Changed default script language into JavaScript

Commit:
Started with the AStar implementation

Commit:
Added support to access the ModelContext, that contains the current execution state, from the model implementation.

Commit:
Updated stop conditions to handle multiple elements with the same name also added Hibernate Validator reference implementation for JSR 349.

nilols committed:
Added validation test for StopCondition (ReachedEdge)

nilols committed:
Bundles the Javadoc documentation into a jar file

nilols committed:
Added the plugin modules

nilols committed:
Moved classes around to simplify dependencies

nilols committed:
Started to update the maven plugin by adding new goals in order to enhance it a bit.

nilols committed:
Chained validate and test mojo so whenever graphwalker:test is called, a validation of models and implementations is performed, before the test execution starts.

nilols committed:
Changed plugin annotations into java 5 version

nilols committed:
GenerateMojo will now generate interface from graphml files, but we need to refactor and take care of all TODO

nilols committed:
Added @Model annotation and simplified GenerateMojo

nilols committed:
Removed the force parameter, wanna keep it simple. We can add more configuration possibilities later on.

nilols committed:
Improved the code generator, models will be converted into interfaces, when executing the generate goal of the maven plugin. Methods not used by the model will be removed, if the interface file already exits. New edges/vertices in the model will result in new methods in the interface.

nilols committed:
Split the generate mojo into generate and test-generate mojos

nilols committed:
Updated the order plugin goals will be executed

nilols committed:
Fixed javadoc tags

nilols committed:
Added groups property, comment in generated files and fix for searching non-existing folders.

nilols committed:
Changed the @GraphWalker annotation to supports multiple @Execution annotations. Each @Execution supports the normal execution parameters and by adding a group definition we can select which executions to run.

nilols committed:
Removed @Execute for the mojos in the maven plugin, except the test mojo, due to that otherwise the process is forked several times during a test execution.

nilols committed:
Changed name on the execution annotation and changed the example implementation to use the new annotations.

nilols committed:
Changed package name for the ShoppingCart model

nilols committed:
Updated the maven plugin, lifecycle and configuration parameters.

nilols committed:
Because, both the jenkins and the maven plugin, needs to be able to read/write reports, a new report module is created. This will also decouple reports from the core.

nilols committed:
Refactor

nilols committed:
Refactor

nilols committed:
Added a flat profiler for computing min, max and average execution times of ModelElements (Edges and Vertices).

nilols committed:
Added implementation for the Amazon test and updated the code generator so that models Start method is not generated.

nilols committed:
Updated the Amazon implementation, changed default script language back to groovy, implementation and model can now easily exchange information through the context.

nilols committed:
Refactor core and updated the example so that the browser closes after test execution.

nilols committed:
Updated annotations

nilols committed:
Removed unused factories

nilols committed:
Removed unused exceptions

nilols committed:
Moved report back to core

nilols committed:
Code generator can now handle unnamed elements

nilols committed:
Added to the model that if the shopping cart already contains books, we first remove them before continue with the rest of the test. The example now reads the num_of_books defined in the model and also updates the same variable based on the number of items in the cart on the Amazon page when starting the test. After updating num_of_books it is also used to control the execution flow.

nilols committed:
Implemented the A* algorithm and the A* path generator.

nilols committed:
A* throws exception when no path is found

nilols committed:
Improved the ReflectionUtils search for a matchable constructor when creating new instances

nilols committed:
Refactor

nilols committed:
Skip creating interfaces if no changes are made to the models

nilols committed:
Refactor, removed unused classes

nilols committed:
Started with the upgrading of the jenkins plugin, the purpose is to display status for each job in a main dashboard view fashion and by clicking a job/test in the dashboard will in the future show that particular job's actual status and/or aggregated data, stacktraces, screenshots or other relevant data.

Commit:
Update jenkins plugin

nilols committed:
Implemented the Squarified Treemap algorithm

nilols committed:
Implemented the Squarified Treemap algorithm

nilols committed:
Implemented the Squarified Treemap algorithm

nilols committed:
Started refactoring the maven plugin and moved generated files default location to a more "mavenized" location. Also updated the graphwalker lifecycle so that validate is performed before generation of source files.

nilols committed:
Added default values for @Model parameters

nilols committed:
Refactoring of the maven plugin

nilols committed:
Improved the way include, exclude, -Dtest, -Dgroups works. -Dtest will override include, exclude properties in the pom file. -Dtest=!*Test will exclude all graphwalker tests that ends with Test. -Dgroups parameter will make it possible to only run specific groups.

nilols committed:
Fixed issues with run configuration

nilols committed:
Removed the type from the @Model annotation, a ModelFactory needs to determine by it self if the defined model is supported or not e.g. check the file extension by it self or open the file and figure out if the content is supported.

nilols committed:
Refactoring of maven goals

nilols committed:
Refactoring of maven mojos

nilols committed:
Fixed problem with ExceptionHandlers not being called

nilols committed:
If no @Execution annotation is added to the @GraphWalker annotation a default execution will be created which make it easier to create a test that follows convention over configuration. Also started with a new test that will utilize this concept.

nilols committed:
Removed the ExceptionStrategy

nilols committed:
Information about selected path generator and stop condition is now printed before the test starts

nilols committed:
Refactor

nilols committed:
Added methods for validating models and now failed test will print out more information about chosen PathGenerator, StopCondition, Value and the fulfilment of the model when the test stopped.

nilols committed:
Fixed logging aspect

nilols committed:
Fixed problem with different tests using the same ElementFactory

nilols committed:
Changed the default stop condition to EdgeCoverage

nilols committed:
Removed report classes from core

nilols committed:
Added files needed to create junit reports

nilols committed:
Removed the filter property from the ExceptionHandler annotation, filtering is now done by adding an argument to the handler method. The specific exception will also be handed to the method.

nilols committed:
Changed the google test to use multiple models

nilols committed:
Divided the google example into multiple models

nilols committed:
Created a new maven module, the intention of the api module is to make it easier to augment graphwalker.

nilols committed:
Changed version

nilols committed:
Updated API

nilols committed:
Updated API

nilols committed:
Updated API

nilols committed:
Updated API

nilols committed:
Updated API

nilols committed:
Updated API

nilols committed:
Updated API

nilols committed:
Updated API

nilols committed:
Updated API

nilols committed:
Updated API

nilols committed:
Updated API

nilols committed:
Updated API

nilols committed:
Updated API

nilols committed:
Started with implementation based on the new API

nilols committed:
Updated API + Core

nilols committed:
Updated API + Core

nilols committed:
Updated API + Core

nilols committed:
Started a new rewrite of the code, I will try keep it simple in the core component but at the same time support as many earlier features as possible and use as few external libraries as possible.

nilols committed:
updated model implementation

nilols committed:
updated model implementation

nilols committed:
Added the DFS implementation

nilols committed:
Added the Floyd Warshall and the A* implementations

nilols committed:
Added default stop condition and path generator files

nilols committed:
Added profiler and started with the Sink interface

nilols committed:
Updated license

nilols committed:
Added ExecutionContext and updated StopCondition and PathGenerator interfaces

nilols committed:
Added implementation for StopConditions

nilols committed:
Added implementations for PathGenerators

nilols committed:
Added implementation for ExecutionContext

nilols committed:
Added the text resource handling

nilols committed:
Implemented visit counter and restart function

nilols committed:
Added annotations

nilols committed:
Updated changed class name and package structure

nilols committed:
Added annotations and changed the machine signature

nilols committed:
Fixed problem with generating interfaces on windows

nilols committed:
Fixed problem with generating interfaces on windows

KristianKarl-zz committed:
Initial work

KristianKarl committed:
Worked on Unit tests

KristianKarl-zz committed:
Created io module

KristianKarl-zz committed:
Changed package name to reflect module name

KristianKarl-zz committed:
Name change

KristianKarl-zz committed:
Added some tests for io

KristianKarl-zz committed:
Added more tests

Commit:
Added tests for path generators

KristianKarl-zz committed:
Working with multiple generators and stop conditions

KristianKarl-zz committed:
Work ing progress with combined generators and multiple stop conditions

KristianKarl-zz committed:
Work in progress with combined generators and multiple stop conditions

Commit:
Added the Eulerian class that can determine which type of graph the actual model is.

Commit:
Changed the compiler source and target levels

Commit:
Added the base of the new graphwalker-soap module

KristianKarl-zz committed:
WIP

Commit:
Added the base of the new graphwalker-webrenderer module

KristianKarl-zz committed:
WIP

Commit:
Updated versions

KristianKarl-zz committed:
Working generator / stop condition CLI parser

Commit:
Started implement the WebRenderer

Commit:
Added license headers

Commit:
Merge branch 'webrenderer'

KristianKarl committed:
WIP change to JCommander

Commit:
Added extra classes for the geometry

Commit:
Added filter for mac specific files

KristianKarl committed:
WIP

KristianKarl committed:
Fixed Offlline

Commit:
Using the graphstream project to layout the graph

Commit:
Changed FileSystemException into a unchecked ModelFactoryException

Commit:
Fixed problem with license headers

Commit:
Got rid of unwanted printouts

Commit:
Removed the commons-vfs2 dependency

Commit:
Added support for opening folders with a lot of graphml files

Commit:
Workaround for a performance issue, regarding flattening of a large amount of smaller models

Commit:
Updated the folder path, vertex and edge count for the largeNumberOfSubGraphs test

Commit:
Remove the test execution from graphwalker-example

KristianKarl committed:
CLI WIP

KristianKarl committed:
Muted some unit tests. They need to be unmuted when gw is feature complete

Commit:
Fix problem with source code generation

Commit:
Add new mojo that automatically updates generated interface

Commit:
Update the watch mojo

Commit:
Update the plugin implementation

Commit:
Javadoc

Commit:
Only valid method names will be used when generating code

Commit:
Add possibility to choose script language

Commit:
Add a helper method for accessing script variables

KristianKarl committed:
Update version of jcommander

KristianKarl committed:
Update unit tests

KristianKarl committed:
Update models used for unit tests

KristianKarl committed:
Typo

KristianKarl-zz committed:
Wrong indentation off help option

KristianKarl-zz committed:
Ignore file

KristianKarl-zz committed:
Initial documentation

KristianKarl-zz committed:
Description

KristianKarl-zz committed:
Refactoring

KristianKarl-zz committed:
Bug fix

KristianKarl-zz committed:
Removed test that was taking too long time to run

KristianKarl-zz committed:
Added modules

KristianKarl-zz committed:
Update from File to String

Commit:
Repack the testng Assert so users doesn't need extra dependencies

Commit:
Update code to java 7

Commit:
Execute @BeforeExecution and @AfterExecution during test execution

KristianKarl-zz committed:
Fixed bug using multiple generators in offline

Commit:
Update the handling of ExecutionStatus

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Fixed id in Amazon test

KristianKarl-zz committed:
Update selenium version

KristianKarl committed:
Initial Travis CI integration

KristianKarl committed:
Trying package command

KristianKarl committed:
Update README.md

Added build badge from Travis-CI

KristianKarl committed:
Change back to install

KristianKarl committed:
Less verbose output

Fix: "The log length has exceeded the limit of 4 Megabytes (this usually means that test suite is raising the same exception over and over)." from travis

KristianKarl committed:
Crap character in file

Copy and paste typo

Commit:
Fixed problem with parsing a Guard on multiple lines

Commit:
Fixed problem with parsing a Guard on multiple lines

Commit:
Merge branch 'master' of https://github.com/nilols/GraphWalker

Commit:
Add handling of guards to the core module

Commit:
Update plugin versions

Commit:
Add plugin to verify that the test coverage meet our expectation

Commit:
Clean up the code

Commit:
Clean up the code

Commit:
Add unit test for the Assert class

Commit:
Add a reference (impl) to the implementation from the script engine

Commit:
Clean up and add more unit tests

Commit:
Move resource classes to the io module

Commit:
Clean up

Commit:
Add @Vertex and @Edge marker annotations

KristianKarl committed:
Only build master branch

Commit:
Update annotations

Commit:
Update license

Commit:
Refactor

Commit:
Handel BLOCKED keyword

Commit:
Merge branch 'master' of https://github.com/nilols/GraphWalker

Commit:
Update README.md
